### PR TITLE
Fixes to resolve skipped tests with ECC disabled

### PR DESCRIPTION
**Pull request by dgarske**
* Fix for macro check on `BUILD_TLS_ECDHE_RSA_WITH_CHACHA20_OLD_POLY1305_SHA256` to check for HAVE_ECC - Thanks Jacob.
* Fix when no cert specified using default, do not return failure in suite tests. Thanks Kaleb.
* Fix for unit.test fails with `-H verifyFail`.
* Fix bug with SendClientKeyExchange and ifdef logic for `ecdhe_psk_kea`, which was preventing `ECDHE-PSK` from working if HAVE_CURVE25519 was defined.
* Disabled broken downgrade test in test-tls13-down.conf (@SparkiDev will need to investigate).
* Various spelling fixes.

### CODE REVIEW

File: src/tls13.c [hunk 1]
```diff
@@ -8450,10 +8450,11 @@ int wolfSSL_write_early_data(WOLFSSL* ssl, const void* data, int sz, int* outSz)
     if (ssl->options.handShakeState == NULL_STATE) {
         ssl->earlyData = expecting_early_data;
         ret = wolfSSL_connect_TLSv13(ssl);
-        if (ret <= 0)
+        if (ret != WOLFSSL_SUCCESS)
```
Comment (toddouska) [inline src/tls13.c]: What case makes this different?  That is, what non 1 positive number wasn't being handled?

File: src/tls13.c [hunk 2]
```diff
@@ -8450,10 +8450,11 @@ int wolfSSL_write_early_data(WOLFSSL* ssl, const void* data, int sz, int* outSz)
...
             return WOLFSSL_FATAL_ERROR;
     }
-    if (ssl->options.handShakeState == CLIENT_HELLO_COMPLETE) {
+    if (ssl->options.handShakeState == CLIENT_HELLO_COMPLETE ||
+        ssl->options.handShakeState == HANDSHAKE_DONE) {
```
Comment (toddouska) [inline src/tls13.c]: Does early data make sense post handshake?

Comment (dgarske) [reply src/tls13.c]: We need @SparkiDev to review this. I sent him an email yesterday, but not response.

Comment (SparkiDev) [reply src/tls13.c]: Early Data is not allowed post-handshake.
It is only allowed during a handshake after ClientHello.

**Pull request merged**